### PR TITLE
feat:(virtualization): first implementation of the virtualization subcommand

### PR DESCRIPTION
**Pull request by yaroslavborbat**
## Description
Add a virtualization subcommand.  Implemented console, port forwarding, vnc, ssh, scp.
Use cases:
```
d8 virtualization console myvm.mynamespace 
d8 virtualziation port-forward myvm.mynamespace 8080
d8 virtualization vnc myvm.mynamespace 
d8 virtualization ssh user@myvm.mynamespace 
d8 virtualization scp file user@test-vm-provisioning.test:/home/user/file
```

### CODE REVIEW

File: internal/virtualization/templates/target.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	if arg[0] == '.' {
+		return "", "", errors.New("expected name before '.'")
+	}
+	if arg[len(arg)-1] == '.' {
+		return "", "", errors.New("expected namespace after '.'")
+	}
+
+	parts := strings.FieldsFunc(arg, func(r rune) bool {
+		return r == '.'
+	})
```
Comment (diafour) [inline internal/virtualization/templates/target.go:42]: strings.Split(arg, ".") ?

File: internal/virtualization/templates/templates.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	return `The following options can be passed to any command:{{if .HasAvailableInheritedFlags}}
+
+{{.InheritedFlags.FlagUsages | trimTrailingWhitespaces}}{{end}}
+`
+}
+
+// ExactArgs validate the number of input parameters
+func ExactArgs(nameOfCommand string, n int) cobra.PositionalArgs {
+	return func(cmd *cobra.Command, args []string) error {
+		if len(args) != n {
```
Comment (diafour) [inline internal/virtualization/templates/templates.go:78]: Use 
```
if len(args) == n {
  return nil
}
```
And unindent next code.

File: internal/virtualization/templates/target.go (hunk 1 above)
Comment (yaroslavborbat) [reply internal/virtualization/templates/target.go:42]: if arg equal "vm...namespace"
with FieldsFunc we have ["vm", "namespace"]
with Split we have ["vm", "", "", "namespace" ]

File: internal/virtualization/templates/templates.go (hunk 2 above)
Comment (yaroslavborbat) [reply internal/virtualization/templates/templates.go]: ++

**Pull request merged**